compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt

```diff
@@ -67,15 +67,19 @@ internal class FlushCoroutineDispatcher(
     }
 
     /**
-     * Whether the dispatcher has any tasks scheduled or currently running.
+     * Whether the dispatcher has immediate tasks pending.
      */
-    fun hasTasks() = synchronized(tasksLock) {
-        immediateTasks.isNotEmpty() || delayedTasks.isNotEmpty()
-    } || isPerformingRun
+    fun hasImmediateTasks() = isPerformingRun ||
+        synchronized(tasksLock) { immediateTasks.isNotEmpty() }
+
+    /**
+     * Whether the dispatcher has delayed tasks pending.
+     */
+    fun hasDelayedTasks() = synchronized(tasksLock) { delayedTasks.isNotEmpty() }
```

Comment:
Wondering if  separate tasksLocks per the tasks list would be beneficial (if tasks are added/consumed from different threads). Or would it create a problem? 
___

Just trying to understand how it works.

Reply:
I don't know whether it will actually be beneficial for performance, but it makes sense, so I changed it.